gap/CompilerLogic.gi

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# FinSetsForCAP: The elementary topos of (skeletal) finite sets
+#
+# Implementations
+#
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ ],
+        src_template := "1 + 1",
+        dst_template := "2",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ ],
+        src_template := "2 + 1",
+        dst_template := "3",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "number" ],
+        src_template := "1 * number",
+        dst_template := "number",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "number" ],
+        src_template := "number * 1",
+        dst_template := "number",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "list" ],
+        src_template := "list{[ ]}",
+        dst_template := "[ ]",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "list" ],
+        src_template := "list{[ 2 ]}",
+        dst_template := "[ list[2] ]",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "entry", "pos" ],
+        src_template := "[ entry ][pos]",
+        dst_template := "entry",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ ],
+        src_template := "Product( [ ] )",
+        dst_template := "1",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "func" ],
+        src_template := "Product( [ ], func )",
+        dst_template := "1",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "entry" ],
+        src_template := "Product( [ entry ] )",
+        dst_template := "entry",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "entry" ],
+        src_template := "Product( [ 1, entry ] )",
+        dst_template := "entry",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "entry1", "entry2" ],
+        src_template := "Product( [ entry1, entry2 ] )",
+        dst_template := "( entry1 * entry2 )",
```

Comment:
```suggestion
        dst_template := "entry1 * entry2",
```

Reply:
Done.

Reply:
Either GitHub is caching something or you have not pushed the change.

Reply:
My mistake.